### PR TITLE
Adding sasktran2

### PR DESCRIPTION
**Pull request by dannyzed**
Adds the SASKTRAN2 radiative transfer model

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dannyzed** [comment]
@conda-forge/help-python-c, ready for review!

**Comment by dannyzed** [comment]
@conda-forge-admin, please ping team

File: recipes/sasktran2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - {{ compiler('fortran') }}  # [not win]
+  host:
+    - python {{ python }}
+    - pip
+    - eigen
+    - libblas =*=*netlib
+    - libcblas =*=*netlib
+    - liblapack =*=*netlib
+    - liblapacke =*=*netlib
+    - libtmglib =*=*netlib
```
Comment (xhochy) [inline recipes/sasktran2/meta.yaml:36]: The global pinning will take care of that.
```suggestion
    - libblas
    - libcblas
    - liblapack
    - liblapacke
    - libtmglib
```

File: recipes/sasktran2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - appdirs
+    - pyyaml
+    - xarray
+    - dask
+    - netcdf4
+    - scipy
+    - libblas
+    - liblapacke
+    - llvm-openmp  # [mac]
+    - libgomp  # [linux]
```
Comment (xhochy) [inline recipes/sasktran2/meta.yaml:57]: They should already come in via run_exports 
```suggestion
```

File: recipes/sasktran2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  host:
+    - python {{ python }}
+    - pip
+    - eigen
+    - libblas =*=*netlib
+    - libcblas =*=*netlib
+    - liblapack =*=*netlib
+    - liblapacke =*=*netlib
+    - libtmglib =*=*netlib
+    - llvm-openmp  # [mac]
```
Comment (xhochy) [inline recipes/sasktran2/meta.yaml:37]:
```suggestion
    - llvm-openmp  # [osx]
```

File: recipes/sasktran2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+requirements:
+  build:
+    - cmake >=3.5
+    - make  # [not win]
+    - python  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}  # [not win]
+  host:
+    - python {{ python }}
```
Comment (xhochy) [inline recipes/sasktran2/meta.yaml:29]:
```suggestion
    - python
```

File: recipes/sasktran2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    - liblapacke =*=*netlib
+    - libtmglib =*=*netlib
+    - llvm-openmp  # [mac]
+    - zlib
+    - pybind11 >=2.10.2
+    - spdlog
+    - scikit-build-core
+    - setuptools-scm
+  run:
+    - python {{ python }}
```
Comment (xhochy) [inline recipes/sasktran2/meta.yaml:44]:
```suggestion
    - python
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**